spec/API_specification/array_api/elementwise_functions.py

```diff
@@ -401,6 +401,34 @@ def ceil(x: array, /) -> array:
         an array containing the rounded result for each element in ``x``. The returned array must have the same data type as ``x``.
     """
 
+def conj(x: array, /) -> array:
+    """
+    Returns the complex conjugate for each element ``x_i`` of the input array ``x``.
+
+    For complex numbers of the form
+
+    .. math::
+       a + bj
+
+    the complex conjugate is defined as
+
+    .. math::
+       a - bj
+
+    Hence, the returned complex conjugates must be computed by negating the imaginary component of each element ``x_i``.
+
+    Parameters
+    ----------
+    x: array
+        input array. Should have a complex-floating point data type.
```

Comment:
@kgryte Could you remind me what's our recommendation for the exception? `ValueError` or `TypeError` (or neither)?

Reply:
Typically, we don't specify the exact form an exception should take and defer to implementing libraries to make that decision.

Reply:
Additionally, the spec here uses SHOULD to allow implementations to support other data types due to, e.g., backward compat concerns. So requiring an exception is not desirable.

Reply:
IMO, I think the restriction is something we still need to decide (both here and for `real` and `imag`; was on the agenda for the most recent consortium meeting, but we ran out of time). Personally, I'm sympathetic to both allowing all numeric dtypes and to limiting to only complex dtypes.

Reply:
+1 no strong opinion here.

Reply:
I would prefer to allow both complex and real inputs for `real`, `imag` and `conj`. This is useful for writing generic routines that handle real or complex numbers.

Reply:
In the 9/22/2022 meeting, the plan is to go ahead and merge the PR as is (i.e., restricted to complex dtypes). We can revisit if implementors receive user feedback that the API should be extended to real-valued dtypes.